### PR TITLE
docs: Deep packet Inspection service to enforce policy rules on 5G SA Deployments

### PR DESCRIPTION
**Pull request by prabinakpattnaik**
Signed-off-by: prabinak <prabinak@wavelabs.ai>

## Summary
The design (proposal) document created for Deep packet Inspection service to enforce policy rules on 5G SA Deployments support. TIP OCN Functional requirement REQ-OCN-18 for FWA deployment asks for application detection as a minimum DPI requirement.

It is provided the proposal document for gap (design) analysis and more details to provide DPI support on magma OVS and Pipelined.
## Additional Information
Created the proposal issue #11134 for this PR.

### CODE REVIEW

**Review by electronjoe — APPROVED** [review]
Per Proposal Process, this design doc has been reviewed strictly for formatting and process / basic typos - Looks Good to Me!

File: docs/readmes/proposals/p025_Deep_packet_Inspection.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+5) Enterprises can leverage DPI to block or throttle access to risky or unauthorized applications, block policy-violating usage patterns or prevent unauthorized data access within corporate-approved applications, stop data exfiltration attempts by external attackers or potential data leaks caused by both malicious and negligent insiders.
+
+Wavelabs as a VAR (Value Added Reseller) for magma 5G SA core received RFPs from FWA service providers. DPI is a feature asked by all the service providers. 
+
+## Gap (Design) Analysis
+To provide DPI support on magma  OVS and Pipelined will need the following changes;
+1.  Provide the new Opaque DPI Interface that integrates with the open source DPI plugin. (dynamically any 3rd party / external / customer DPI engine can be hooked with OVS seamlessly)
+2.  Implement the sample plugin will demonstrate the DPI public API that was given for DPI-enabled-OVS. This sample plugin will simply write out the ethernet packet from OVS to a file.
```
Comment (koolzz) [inline docs/readmes/proposals/p025_Deep_packet_Inspection.md:44]: This thing looks ancient https://github.com/kspviswa/dpi-enabled-ovs, any idea if it'll work with our ovs version? Or are you intending to update it?

File: docs/readmes/proposals/p025_Deep_packet_Inspection.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+3.  Implement the test-dpi plugin, to demonstrate the interfacing capablity of DPI with OVS. This can be extended to full-fledged DPI engine as we now have raw ethernet packet from OVS
+4.  Implement the DPI engine
```
Comment (koolzz) [inline docs/readmes/proposals/p025_Deep_packet_Inspection.md:46]: I'm assuming no one will be actually Implementing the DPI engine, so lets just say `Setup a DPI engine?`

**Comment by koolzz** [comment]
From our previous DPI deployments a large issue was installing the classifier flows in OVS for policy enforcement based on app/service type(tagging the unique flows with specified ID classifier produced by the DPI engine). We had to send a grpc msg to pipelined with the flow details/which would then install the flow in OVS, this was causing a lot of overhead at large numbers of unique flows. Do we have any plan to tackle that? Installing flows from the DPI engine itself should improve things slightly.

Per my understanding the dpi ovs plugin doesn't help with this as it only facilitates the classification of traffic by forwarding it over to the DPI engine correct?

**Comment by ekowtaylor** [comment]
@prabinakpattnaik do you still plan to move forward with this change? If yes then please reopen and rebase.

**Pull request closed without merging**